build.sc

```diff
@@ -182,80 +126,131 @@ trait ChiselPublishModule extends CiReleaseModule {
 
 }
 
-object firrtl extends Cross[Firrtl](v.scalaCrossVersions)
-
-trait Firrtl extends common.FirrtlModule with CrossSbtModule with ScalafmtModule {
-  def millSourcePath = super.millSourcePath / os.up / "firrtl"
+trait HasScala2MacroAnno extends CrossSbtModule {
+  override def scalacOptions = T {
+    if (!utils.isScala3(crossValue)) {
+      super.scalacOptions() ++ Agg("-Ymacro-annotations")
+    } else super.scalacOptions()
+  }
+}
 
-  override def scalacOptions = v.commonOptions ++ Seq(
-    "-language:reflectiveCalls",
-    "-language:existentials",
-    "-language:implicitConversions",
-    "-Yrangepos", // required by SemanticDB compiler plugin
-    "-Xsource:3",
-    "-Xsource-features:infer-override"
-  )
+trait HasScala2Plugin extends CrossSbtModule {
+  def pluginModule: Plugin
 
-  def osLibModuleIvy = v.osLib
+  override def scalacOptions = T {
+    if (!utils.isScala3(crossValue)) {
+      super.scalacOptions() ++ Agg(s"-Xplugin:${pluginModule.jar().path}")
+    } else super.scalacOptions()
+  }
 
-  def json4sIvy = v.json4s
+  override def scalacPluginClasspath = T {
+    if (!utils.isScala3(crossValue)) {
+      super.scalacPluginClasspath() ++ Agg(pluginModule.jar())
+    } else super.scalacPluginClasspath()
+  }
+}
 
-  def dataclassIvy = v.dataclass
+object firrtl extends Cross[Firrtl](v.scalaCrossVersions)
 
-  def commonTextIvy = v.commonText
+trait Firrtl extends CrossSbtModule with Cross.Module[String] with HasScala2MacroAnno with ScalafmtModule {
+  def millSourcePath = super.millSourcePath / os.up / "firrtl"
+  def scalaVersion = crossValue
+
+  override def scalacOptions = T {
+    utils.isScala3(crossValue) match {
+      case false => v.scala2CommonOptions ++ Seq(
+        "-language:reflectiveCalls",
+        "-language:existentials",
+        "-language:implicitConversions",
+        "-Yrangepos", // required by SemanticDB compiler plugin
+        "-Xsource:3",
+        "-Xsource-features:infer-override"
+      )
+      case true => Seq.empty[String]
+    }
```

Comment:
This is definitely a nit, but it's a bit weird to match on Boolean values vs. just good 'ole if-else (also note use of crossScalaValue) which is a safer option to use.
```suggestion
    if (utils.isScala3(crossScalaValue)) {
      Seq.empty[String]
    } else {
      v.scala2CommonOptions ++ Seq(
        "-language:reflectiveCalls",
        "-language:existentials",
        "-language:implicitConversions",
        "-Yrangepos", // required by SemanticDB compiler plugin
        "-Xsource:3",
        "-Xsource-features:infer-override"
      )
    }
```

Reply:
yeah true, will change this